lib/prometheus-parser/src/lib.rs

```diff
@@ -67,7 +67,19 @@ vector_common::impl_event_data_eq!(ParserError);
 #[derive(Debug, Eq, Hash, PartialEq)]
 pub struct GroupKey {
     pub timestamp: Option<i64>,
-    pub labels: BTreeMap<String, String>,
+    pub labels: Vec<(String, String)>,
```

Comment:
Are there going to be any behavioral differences here between BTreeMap and Vec? Ordering/insert speed/lookup speed/etc

Reply:
The `BTreeMap` can contain only one value for any given key. The `Vec` can contain any number of either.